digest/src/lib.rs

```diff
@@ -273,6 +273,21 @@ pub trait VariableOutputReset: VariableOutput + Reset {
     }
 }
 
+/// Trait for hash functions with personalization string for domain separation.
+pub trait PersonalizationInit: Sized {
+    // It would be nice to define a default value equal to `[u8]`, but unfortunately
+    // associated type defaults are currently unstable
+
+    /// Extended personalization
+    type PersonalizationExtArg;
+
+    /// Create new hasher instance with the given personalization string.
+    fn new(personalization: &[u8]) -> Self;
```

Comment:
Is this name going to conflict with other traits that define a `new` method that might be impl'd for the same type?

Reply:
Yeah, it's probably worth to rename it to something like `new_personilized`.